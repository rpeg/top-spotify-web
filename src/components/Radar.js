import React from 'react';
import Chart from 'react-apexcharts';

const getFeatureAvg = (features, name) => {
  const featureArr = features.map((f) => f[name]);
  const total = featureArr.reduce((acc, curr) => acc + curr);

  return total / features.length;
};

const Radar = ({ features }) => {
  const labels = [
    'acousticness',
    'danceability',
    'energy',
    'instrumentalness',
    'liveness',
    'speechiness',
    'valence',
  ];

  const options = {
    labels,
    yaxis: {
      min: 0,
      max: 1,
      show: false,
    },
    chart: {
      foreColor: '#FFFFFF',
      fontFamily: 'Karla, sans-serif',
      toolbar: {
        show: false,
      },
    },
  };

  const series = [
    {
      data: labels.map((l) => getFeatureAvg(features, l)),
    },
  ];

  return (
    <div>
      <Chart type="radar" options={options} series={series} />
    </div>
  );
};

export default Radar;
