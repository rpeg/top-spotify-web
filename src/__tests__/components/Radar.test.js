import React from 'react';
import { shallow } from 'enzyme';

import Radar from '../../components/Radar';

describe('Radar', () => {
  const features = [];

  it('should not log errors in console', () => {
    const wrapper = shallow(<Radar features={features} />);
    const spy = jest.spyOn(global.console, 'error');
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('renders correctly', () => {
    const wrapper = shallow(<Radar features={features} />);
    expect(wrapper).toMatchSnapshot();
  });
});
