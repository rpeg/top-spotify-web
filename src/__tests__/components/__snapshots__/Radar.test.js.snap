// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radar renders correctly 1`] = `
<div>
  <Radar
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": "rgba(29, 185, 84, 0.1)",
            "borderColor": "rgba(29, 185, 84, 1)",
            "data": Array [
              NaN,
              NaN,
              NaN,
              NaN,
              NaN,
              NaN,
              NaN,
            ],
            "label": undefined,
            "pointBackgroundColor": "rgba(29, 185, 84,1)",
            "pointBorderColor": "#fff",
            "pointHoverBackgroundColor": "#fff",
            "pointHoverBorderColor": "rgba(29, 185, 84,1)",
          },
        ],
        "labels": Array [
          "acousticness",
          "danceability",
          "energy",
          "instrumentalness",
          "liveness",
          "speechiness",
          "valence",
        ],
      }
    }
    options={
      Object {
        "gridLines": Object {
          "display": false,
        },
        "legend": Object {
          "display": false,
        },
        "scale": Object {
          "gridLines": Object {
            "color": "rgba(255, 255, 255, .5)",
          },
          "pointLabels": Object {
            "fontColor": "white",
            "fontFamily": "Karla, sans-serif",
            "fontSize": 12,
          },
          "ticks": Object {
            "beginAtZero": true,
            "display": false,
            "max": 1,
            "min": 0,
            "stepSize": 1,
          },
        },
      }
    }
  />
</div>
`;
